tests/upgrade_path/upgrade_helpers.py

```diff
@@ -59,6 +60,40 @@ def check_sonic_version(duthost, target_version):
         "Upgrade sonic failed: target={} current={}".format(target_version, current_version)
 
 
+def restore_image_to_first_boot(duthost, image_version):
```

Comment:
This new mechanism should be optionally enabled. For folks in community who still want to use the old way (without the hack here).

Reply:
This mechanism will only work if `--base_image_version_list` or `--target_image_version_list` are passed in. Otherwise, current behavior is used.